samcli/lib/build/app_builder.py

```diff
@@ -279,7 +279,7 @@ def update_template(
         for logical_id, resource in template_dict.get("Resources", {}).items():
 
             full_path = get_full_path(stack.stack_path, logical_id)
-            has_build_artifact = full_path in built_artifacts
+            has_build_artifact = full_path in built_artifacts and built_artifacts[full_path]
```

Comment:
Should we change `built_artifacts[full_path]` to `built_artifacts.get(full_path)` for not getting any exception if `full_path` is not in the dict?

Reply:
the first half of the condition is to check that full path in the built_artifacts 
`full_path in built_artifacts`

Reply:
I see, we can also simplify it by replacing whole line with `bool(built_artifacts.get(full_path))` but leave it to you